src/services/item/plugins/published/repositories/itemPublished.ts

```diff
@@ -39,23 +40,23 @@ export const ItemPublishedRepository = AppDataSource.getRepository(ItemPublished
   },
 
   // return public item entry? contains when it was published
-  async getAllPublishedItems(): Promise<Item[]> {
-    // we get the nested relation of item.creator because we only return the item and without this the creator is not returned
-    const publishedRows = await this.createQueryBuilder()
-      .select(['item'])
-      .from(Item, 'item')
+  async getAllPublishedItems(itemRepository: typeof ItemRepository): Promise<Item[]> {
+    // we need to pass through item repository because typeorm deduces the entity from its original repository
+    const publishedRows = await itemRepository
+      .createQueryBuilder('item')
       .leftJoinAndSelect('item.creator', 'creator')
       .innerJoin('item_published', 'ip', 'ip.item_path = item.path')
       .getMany();
+
     return publishedRows;
   },
 
   // return public item entry? contains when it was published
-  async getItemsForMember(memberId: string) {
+  async getItemsForMember(itemRepository: typeof ItemRepository, memberId: string) {
```

Comment:
At this point I might consider moving the function directly in ItemRepository `getPublishedItemsForMember`, because I often expect a `X` Entity when I call a `XRepository`.
Most of these queries are basically using item published table as a filter but return `Item`